packages/flutter_tools/test/commands.shard/hermetic/attach_test.dart

```diff
@@ -727,20 +766,20 @@ class FakeAndroidDevice extends Fake implements AndroidDevice {
   bool isSupportedForProject(FlutterProject flutterProject) => true;
 
   @override
-  DevicePortForwarder portForwarder;
+  DevicePortForwarder? portForwarder;
 
-  DeviceLogReader Function() onGetLogReader;
+  DeviceLogReader Function()? onGetLogReader;
 
   @override
   FutureOr<DeviceLogReader> getLogReader({
-    AndroidApk app,
+    AndroidApk? app,
     bool includePastLogs = false,
   }) {
-    return onGetLogReader();
+    return onGetLogReader!();
```

Comment:
by adding a bang, tests that don't hit this code path don't have to inject a fake that never gets used, but if after a refactor it does get used, the test will fail forcing the developer to now inject a fake in the constructor.

Reply:
Did you mean for this to behave differently from the `UnimplementedError` you added in `FakeIOSDevice`?  I think I like that better to make it clearer.

Reply:
SGTM

Reply:
done